recovery/mainwindow.cpp

```diff
@@ -951,29 +954,48 @@ void MainWindow::on_list_doubleClicked(const QModelIndex &index)
     }
 }
 
-void MainWindow::startNetworking()
+void MainWindow::copyWpa()
 {
+    //This file is the one used by dhcpcd
     QFile f("/settings/wpa_supplicant.conf");
-
     if ( f.exists() && f.size() == 0 )
     {
         /* Remove corrupt file */
         f.remove();
     }
-    if ( !f.exists() )
+
+    /* If user supplied a wpa_supplicant.conf on the FAT partition copy that one to settings regardless */
+    if (QFile::exists("/mnt/wpa_supplicant.conf"))
     {
-        /* If user supplied a wpa_supplicant.conf on the FAT partition copy that one to settings
-           otherwise copy the default one stored in the initramfs */
-        if (QFile::exists("/mnt/wpa_supplicant.conf"))
-            QFile::copy("/mnt/wpa_supplicant.conf", "/settings/wpa_supplicant.conf");
-        else
-        {
-            qDebug() << "Copying /etc/wpa_supplicant.conf to /settings/wpa_supplicant.conf";
-            QFile::copy("/etc/wpa_supplicant.conf", "/settings/wpa_supplicant.conf");
-        }
+        qDebug() << "Copying  user wpa_supplicant.conf to /settings/wpa_supplicant.conf";
+
+        QProcess::execute("mount -o remount,rw /settings");
+        QProcess::execute("mount -o remount,rw /mnt");
+
+        QFile::remove("/settings/wpa_supplicant.conf.bak");
+        QFile::rename("/settings/wpa_supplicant.conf","/settings/wpa_supplicant.conf.bak");
+        QFile::copy("/mnt/wpa_supplicant.conf", "/settings/wpa_supplicant.conf");
+        f.setPermissions( QFile::WriteUser | QFile::ReadGroup | QFile::ReadOther | QFile::ReadUser );
+
+        /* rename the user file to avoid overwriting any manually set SSIDs */
```

Comment:
IMHO this comment is a bit confusing - is it essentially the same as saying "rename the user file to indicate that it has been copied (and prevent it being re-copied next time)" ?

Although I guess this breaks the "NOOBS prime directive" of not modifying any files on the RECOVERY partition. Although I've not been keeping up to date with NOOBS for a while, so I dunno if that prime-directive is still in place? ;)

Reply:
Yes. I was concentrating more on the consequences of it being copied a second time.

Dunno either, but I wasn't too concerned about that in PINN. Not sure if there's another way around it.

Reply:
I guess with having the same configuration file stored / copied / edited in two different places, there'll never be a "perfect" way of doing things, so it's probably one of those "pick the least worst" situations ;)